prefq/examples/imitation_preference_comparisons.py

```diff
@@ -5,27 +5,194 @@
+        print("\nPrefqGatherer: sending videos to server...")
+
+        left_filename = f"{query_id}-left.webm"
+        right_filename = f"{query_id}-right.webm"
+
+        left_filepath = os.path.join(self.video_dir, left_filename)
+        right_filepath = os.path.join(self.video_dir, right_filename)
+
+        print("\nLeft Filepath:")
+        print(left_filepath)
+
+        with open(left_filepath, "rb") as left_file, open(
+            right_filepath, "rb"
+        ) as right_file:
+            # Read the file data into memory
+            left_video_data = left_file.read()
+            right_video_data = right_file.read()
+
+        payload = {
+            # Send filenames as .json
+            "left_filename": (
+                json.dumps(left_filename),
+                "application/json",
+            ),
+            "right_filename": (
+                json.dumps(right_filename),
+                "application/json",
+            ),  
+            # Use the file data directly
+            "left_video": (
+                left_filename,
+                left_video_data,
+                "application/octet-stream",
+            ),
+            "right_video": (
+                right_filename,
+                right_video_data,
+                "application/octet-stream",
+            ),
+        }
+
+        response = requests.post(self.server_url + "videos", files=payload, timeout=10)
+
+        if response.status_code >= 200 % response.status_code < 400:
+            print("PrefqGatherer: Payload transferred")
+        else:
+            print("PrefqGatherer: Error while sending videos")
+            print(response.status_code)
+
+        print("PrefqGatherer: ...videos sent to server\n")
+
+
+    def _get_feedback_from_server(self):
+        """
+        GET-Request: Receive client feedback from Server
+        
+            After rendering all videos, the PrefQGatherer enters
+            a blocking while loop, until the server has received
+            all feedback data from the Feedback Client.
+        """
+
+        print("\n\PrefqGatherer: Starting request_feedback() [...]")
+        
+        def _wait_for_feedback_request(url):
+            while True:
+                time.sleep(5)
+                response = requests.get(url)
+                if response.status_code == 200:
+                    feedback_data = response.json()
+                    if feedback_data == {}:
+                        print("Query Client: Waiting for feedback...")
+                        continue
+                    else: 
+                        print("Query Client: Feedback received")
+                        break
+            return feedback_data
+
+        feedback_data = _wait_for_feedback_request(self.server_url + "feedback")
+        return feedback_data
+
+
+SERVER_URL = "http://127.0.0.1:5000/"
+
 rng = np.random.default_rng(0)
+video_dir = tempfile.mkdtemp(prefix="videos_")
+print(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
Should we keep this print? If so, please be a bit more verbose to clarify what is being printed to make the logs more readable.

Reply:
just a relict, just removed it